openedx/core/djangoapps/content_tagging/rest_api/v1/views.py

```diff
@@ -61,6 +67,79 @@ def perform_create(self, serializer):
         user_admin_orgs = get_admin_orgs(self.request.user)
         serializer.instance = create_taxonomy(**serializer.validated_data, orgs=user_admin_orgs)
 
+    @action(detail=False, url_path="import", methods=["post"])
+    def create_import(self, request: Request, **_kwargs) -> HttpResponse:
+        """
+        Creates a new taxonomy and imports the tags from the uploaded file.
+        """
+        perm = "oel_tagging.import_taxonomy"
+        if not request.user.has_perm(perm):
+            raise PermissionDenied("You do not have permission to import taxonomies")
+
+        body = TaxonomyImportNewBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        taxonomy_name = body.validated_data["taxonomy_name"]
+        taxonomy_description = body.validated_data["taxonomy_description"]
+        file = body.validated_data["file"].file
+        parser_format = body.validated_data["parser_format"]
+
+        # ToDo: This code is temporary
+        # In the future, the orgs parameter will be defined in the request body from the frontend
+        # See: https://github.com/openedx/modular-learning/issues/116
+        if oel_tagging_rules.is_taxonomy_admin(request.user):
+            orgs = None
+        else:
+            orgs = get_admin_orgs(request.user)
+
+        taxonomy = create_taxonomy(taxonomy_name, taxonomy_description, orgs=orgs)
+        try:
+            import_success = import_tags(taxonomy, file, parser_format)
+
+            if import_success:
+                return HttpResponse(status=200)
+            else:
+                import_error = get_last_import_log(taxonomy)
+                taxonomy.delete()
+                return HttpResponseBadRequest(import_error)
+        except ValueError as e:
+            return HttpResponseBadRequest(e)
```

Comment:
I'm not sure about this one @pomegranited.

We are preventing some code duplication in the `View` but duplicating the logic of the `edx-platform/create_taxonomy`.

It will be a pain if our `create_taxonomy` gets complicated. Or we will get in trouble if we change some behavior from the `oel_tagging/create_taxonomy`.

Do you think it is ok to leave the duplicated `create_import`?

Reply:
I like @pomegranited's suggestion; it re-uses more code. I generally don't like having a separate "create_import" function at all, and usually I'd just have an import function that accepts a blank ID. It's fine to have this function here but we should aim to keep it as small as possible, just calling the other create and import functions.

The content_tagging version of `create_taxonomy` is [only two lines of code](https://github.com/openedx/edx-platform/blob/4eb0ad23f53494367749e3161946770e38af29e9/openedx/core/djangoapps/content_tagging/api.py#L36-L37) so it's fine to duplicate it here. Just make sure to add comments explaining everything and any concerns about that.

Reply:
Thank you for the discussion! :smile: 

Done here: 1b65728e79ffb5f0e410dad30f5c544569acdce7